### PR TITLE
[rds] Adds RDS version scraper

### PR DESCRIPTION
**Pull request by captn3m0**
This is a Python rewrite of the code at
https://github.com/endoflife-date/release-data/pull/83

I gave a shot at using the [GitHub markdown](https://github.com/awsdocs/amazon-rds-user-guide/blob/main/doc_source/MySQL.Concepts.VersionMgmt.md) file for this, but the postgresql page isn't there.

This PR slightly changes the version detection, and makes the date parser more strict.

TODO:

- [x] Add logging, similar to how we have in the other scripts.

### CODE REVIEW

**Review by marcwrobel — APPROVED** [review]
LGTM

File: src/rds.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    releases = {}
+
+    with urllib.request.urlopen(url, data=None, timeout=5) as contents:
+        html = contents.read().decode("utf-8")
+        soup = BeautifulSoup(html, features="html5lib")
+
+        for table in soup.find_all("table"):
+            for row in table.find_all("tr"):
+                columns = row.find_all("td")
+                if len(columns) == 4:  # Only match the 'Supported XXX minor versions' table
```
Comment (marcwrobel) [inline src/rds.py:28]: Mmmh, think I should not have changed this condition. In order to have the exact release date we must also match the "Supported xxx major versions" table. Will fix it.

**Review by marcwrobel — APPROVED** [review]
LGTM now, @captn3m0 I let you do a final review.

File: src/rds.py (hunk 1 above)
Comment (marcwrobel) [reply src/rds.py]: Fixed.

**Review by marcwrobel — APPROVED** [review]
Just did a review and made a small change. LGTM, merging.

**Review by marcwrobel — APPROVED** [review]

**Comment by CodaBool** [comment]
@marcwrobel Hello, I've been away for a bit and catching up on some of this. Thanks for improving the script and reducing dependencies. 


I am a bit confused as to what the purpose of this repository is. I was hoping this script could be used for getting the major and minor version data. However, currently this script is set up to only grab the major versions. 

I know we decided on for the endoflife.date repo to only use major. I thought maybe this repo the minors could be scraped as well. The [last comment](https://github.com/endoflife-date/endoflife.date/pull/2624#issuecomment-1490551127) I see on the matter is that it was TBD. 

Maybe I am missing something about how this repo works. Is this script intended to update the data in the endoflife.date repo? Or is this meant to create a separate API? If it does create a separate API I would want the minor versions to help me at my work. If this is just for updating the endoflife.date repo then I understand why not including minor versions is what you are going for.

Any info on this would be appreciated.

**Comment by marcwrobel** [comment]
Hi @CodaBool, the purpose of this repository is to grab all releases and store them in JSON files in the `release-data` directory, for later consumption by endoflife.date :

- https://github.com/endoflife-date/release-data/blob/main/releases/amazon-rds-postgresql.json
- https://github.com/endoflife-date/release-data/blob/main/releases/amazon-rds-mysql.json

Those files are then used by the https://github.com/endoflife-date/endoflife.date/blob/master/_auto/latest.py script to determine and update the latest release for each release cycle.

So, to summarize this script is only to update the data in the endoflife.date repository :

- JSON files in https://github.com/endoflife-date/release-data/blob/main/releases/ contains all the versions along with their release date.
- Only the latest release for each cycle is used on https://endoflife.date/ (and its API).

> For this reason production databases trail behind on minors until they near deprecation. Then are updated in one go to latest or close to latest in the case of MySQL.

Indeed, you clearly need the minor (for PostgreSQL) and patch (for MySQL) versions along with their EOL date. Could you file two issues (one for MySQL and one for PostgreSQL) on https://github.com/endoflife-date/endoflife.date with what you said in your EDIT so that we switch to minor / patch (if releases are not too frequent) ?

**Comment by marcwrobel** [comment]
> MySQL does not even recommend using the latest minor version

That's an interesting information, even for https://endoflife.date/mysql. Do you have a link documenting this statement ?

**Pull request merged**